src/Interpreters/ErrorLog.cpp

```diff
@@ -0,0 +1,127 @@
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeDateTime64.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Interpreters/ErrorLog.h>
+#include <base/getFQDNOrHostName.h>
+#include <Common/DateLUTImpl.h>
+#include <Common/ThreadPool.h>
+#include <Common/ErrorCodes.h>
+
+#include <vector>
+
+namespace DB
+{
+
+ColumnsDescription ErrorLogElement::getColumnsDescription()
+{
+    return ColumnsDescription {
+        {"hostname", std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()), "Hostname of the server executing the query."},
+        {"event_date", std::make_shared<DataTypeDate>(), "Event date."},
+        {"event_time", std::make_shared<DataTypeDateTime>(), "Event time."},
+        {"code", std::make_shared<DataTypeInt32>(), "Error code}"},
+        {"error", std::make_shared<DataTypeLowCardinality>(std::make_shared<DataTypeString>()), "Error name."},
+        {"value", std::make_shared<DataTypeUInt64>(), "Number of errors happened in time interval."},
+        {"remote", std::make_shared<DataTypeUInt8>(), "Remote exception (i.e. received during one of the distributed queries)."}
```

Comment:
Let's use compression for these columns, take a look at `getColumnsDescription` in AsynchronousMetricLog.cpp.